node/src/reactor/joiner.rs

```diff
@@ -213,15 +213,17 @@ impl From<StorageRequest> for Event {
 
 impl From<NetworkRequest<NodeId, Message>> for Event {
     fn from(request: NetworkRequest<NodeId, Message>) -> Self {
-        Event::SmallNetwork(small_network::Event::from(request))
+        Event::SmallNetwork(small_network::Event::NetworkRequest {
+            req: Box::new(request),
+        })
     }
 }
 
 impl From<NetworkRequest<NodeId, gossiper::Message<GossipedAddress>>> for Event {
     fn from(request: NetworkRequest<NodeId, gossiper::Message<GossipedAddress>>) -> Self {
-        Event::SmallNetwork(small_network::Event::from(
-            request.map_payload(Message::from),
-        ))
+        Event::SmallNetwork(small_network::Event::NetworkRequest {
+            req: Box::new(request.map_payload(Message::from)),
```

Comment:
Same Spiel as above, we're nesting types, but we do not want to box each time there if we can help it.

Reply:
Implemented specific `impl From`